Examples/DemonsRegistration1/DemonsRegistration1.R

```diff
@@ -0,0 +1,66 @@
+#=========================================================================
+#
+#  Copyright Insight Software Consortium
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#=========================================================================
+# Run with:
+#
+# Rscript --vanilla DemonsRegistration1.R fixedImageFilter movingImageFile outputTransformFile
+#
+
+library(SimpleITK)
+
+commandIteration <- function(filter)
+{
+     res <- function() {
+     msg <- paste("Iteration number ", filter$GetElapsedIterations(),
+                  " = ", filter$GetMetric(), "\n" )
+     cat(msg)
+    }
+    return(res)
+}
+
+args <- commandArgs( TRUE )
+
+if (length(args) != 3) {
+    stop("3 arguments expected - fixedImageFilter, movingImageFile, outputTransformFile")
+}
+
+fixed <- ReadImage(args[[1]], 'sitkFloat32')
+
+moving <- ReadImage(args[[2]], 'sitkFloat32')
+
+
+matcher <- HistogramMatchingImageFilter()
+matcher$SetNumberOfHistogramLevels(1024)
+matcher$SetNumberOfMatchPoints(7)
+matcher$ThresholdAtMeanIntensityOn()
+moving <- matcher$Execute(moving, fixed)
+
+demons <- DemonsRegistrationFilter()
```

Comment:
State that this is the "basic" demons algorithm and that there is a family of such algorithms in SimpleITK.